src/modules/deal-folder/dto/upload-file-in-deal-folder-request.dto.ts

```diff
@@ -0,0 +1,42 @@
+import { ENUMS } from '@ukef/constants';
+import { DocumentTypeEnum } from '@ukef/constants/enums/document-type';
+import { EXAMPLES } from '@ukef/constants/examples.constant';
+import { ValidatedStringApiProperty } from '@ukef/decorators/validated-string-api-property.decorator';
+
+export type UploadFileInDealFolderRequestDto = UploadFileInDealFolderRequestItem[];
+
+export class UploadFileInDealFolderRequestItem {
+  @ValidatedStringApiProperty({
+    description: 'The name of the buyer used in the deal.',
+    minLength: 1,
+    maxLength: 250,
+    pattern: /^[\w\-.()\s]+/,
```

Comment:
@avaitonis Sharepoint automatically strips folder and files names of leading or trailing spaces and trailing full stops `.`, causing the querying the list for the newly created file step to fail, as the actual name of the created file path doesn't match what's expected. I see 2 possible solutions:

1. Disallow these character combinations in the request
2. Strip leading or trailing spaces and trailing full stops from what we expect the file path to be before querying the list

Which do you think is best @avaitonis?

Reply:
Both options work for me. @abhi-markan  what is your preference? maybe Portal has some filename validation?

Reply:
(1) sounds good to me, but would be good to know what @abhi-markan thinks too

Reply:
We don't have any file name validation (as far I am aware) in DTFS, perhaps best we go with option 1 and as part of the DTFS migration to eStore we can add these restriction at the UI level.

Reply:
we don't have a file validation, but we do strip (as much as possible) special characters from a file. This is what DTFS uses to format the file names for SP 
```js
const formatFilenameForSharepoint = (filenameWithExtension) => {
  const [extension] = filenameWithExtension.match(/\.[^/.]+/g).reverse();
  const filenameWithoutExtension = filenameWithExtension.replace(/\.[^/.]+$/, '');
  const sanitisedFilename = filenameWithoutExtension.replace(/[\W_]+/g, '_');

  return `${sanitisedFilename}${extension}`;
};
```

Reply:
I've added regex to implement 1) just to be safe